src/extension/src/Utility.py

```diff
@@ -68,3 +69,31 @@ def get_datetime_from_str(date_str):
     def get_str_from_datetime(date):
         return date.strftime(Constants.UTC_DATETIME_FORMAT)
 
+    @staticmethod
+    def extract_version(path):
+        """
+        Extract the version part from a given path.
+        Input: /var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-1.2.5/config
+        Return: 1.2.5
+        """
+        match = re.search(r'([\d]+\.[\d]+\.[\d]+)', path)
+        return match.group(1) if match else ""
+
+    @staticmethod
+    def sort_versions(paths):
+        """
+        Sort paths based on version numbers extracted from paths.
+        Input:
+            ["Microsoft.CPlat.Core.LinuxPatchExtension-1.21.1001",
+            "Microsoft.CPlat.Core.LinuxPatchExtension-1.6.100",
+            "Microsoft.CPlat.Core.LinuxPatchExtension-1.21.100"]
+        Return:
+            ["Microsoft.CPlat.Core.LinuxPatchExtension-1.21.1001",
+            "Microsoft.CPlat.Core.LinuxPatchExtension-1.21.100",
+            "Microsoft.CPlat.Core.LinuxPatchExtension-1.6.100"]
+        """
+        def version_key(path):
+            version_numbers = Utility.extract_version(path)
+            return tuple(map(int, version_numbers.split('.'))) if version_numbers else (0,0,0)
+
```

Comment:
Duplicate code in CoreUtility and and Utility. Please remove one and refer other

Reply:
logic exists in both core and handler